test/e2e/service_test.go

```diff
@@ -81,3 +82,60 @@ func (data *TestData) createClusterIPServiceAndBackendPods(t *testing.T, name st
 
 	return svc, cleanup
 }
+
+// TestNodePortWindows tests NodePort Service on Windows Node. It is a temporary test to replace upstream Kubernetes one:
+// https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/test/e2e/windows/service.go#L42
+func TestNodePortWindows(t *testing.T) {
+	skipIfNoWindowsNodes(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	svcName := "agnhost"
+	if len(clusterInfo.windowsNodes) == 0 {
+		t.Fatalf("The cluster should have at least one Windows Node")
+	}
+	svcNode := nodeName(clusterInfo.windowsNodes[0])
+	svc, cleanup := data.createAgnhostServiceAndBackendPods(t, svcName, svcNode, corev1.ServiceTypeNodePort)
+	defer cleanup()
+	t.Logf("%s Service is ready", svcName)
+
+	clientName := "agnhost-client"
+	require.NoError(t, data.createAgnhostPodOnNode(clientName, nodeName(0)))
+	defer data.deletePodAndWait(defaultTimeout, clientName)
+	_, err = data.podWaitForIPs(defaultTimeout, clientName, testNamespace)
+	require.NoError(t, err)
+
+	nodeIP := clusterInfo.nodes[0].ip
+	nodePort := int(svc.Spec.Ports[0].NodePort)
+	addr := fmt.Sprintf("http://%s:%d", nodeIP, nodePort)
+
+	cmd := append([]string{"curl", "--connect-timeout", "1", "--retry", "5", "--retry-connrefused"}, addr)
```

Comment:
`curl` is available in agnhost? why not use `/agnhost connect`?

Reply:
Yes. The reason is that 1) upstream uses `curl`; 2) `curl` provides many options while `/agnhost connect` has only `--timeout duration   Maximum time before returning an error`.
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/test/e2e/windows/service.go#L77

Reply:
Thanks for clarifying for me